src/mem3/src/mem3_util.erl

```diff
@@ -509,6 +507,59 @@ sort_ranges_fun({B1, _}, {B2, _}) ->
     B1 =< B2.
 
 
+get_or_create_db(DbName, Options) ->
+    case couch_db:open_int(DbName, Options) of
+        {ok, _} = OkDb ->
+            OkDb;
+        {not_found, no_db_file} ->
+            try
+                DbOpts = case mem3:dbname(DbName) of
+                    DbName  -> [];
+                    MDbName -> mem3_shards:opts_for_db(MDbName)
+                end,
+                %% Options1 = opts_merge(DbOpts, Options)
+                Options1 = DbOpts ++ [{create_if_missing, true} | Options],
+                couch_db:open_int(DbName, Options1)
+            catch error:database_does_not_exist ->
+                {error, missing_target}
+            end;
+        Else ->
+            Else
+    end.
+
+
+opts_merge(L1, L2) ->
+    lists:reverse(opts_merge(lists:sort(L1), lists:sort(L2, []))).
+
+opts_merge([], [], Acc) ->
+    Acc;
+opts_merge(L1, [], Acc) ->
+    L1 ++ Acc;
+opts_merge([], R1, Acc) ->
+    R1 ++ Acc;
+opts_merge([A1|R1], [A2|R2], Acc) when is_atom(A1) orelse is_atom(A2) ->
```

Comment:
The complexity here for what's going on seems a bit out of balance. Why not just fold new options and lists:keystore them into the existing option accumulator?

Reply:
Because I wasn't sure how that would work with atom options like `sys_db`? I don't want to include that `opts_merge` code, but we need a merge function that prioritizes items in the first list and can handle atom label options. If you've got a chunk of code to do that then I'll happily use it.

Reply:
```erlang
merge_opts([], Opts) ->
    Opts;
merge_opts([{K, V} | RestToAdd], Opts) ->
    merge_opts(RestToAdd, lists:keystore(K, 1, Opts, {K, V}));
merge_opts([K | RestToAdd], Opts) when is_atom(K) ->
    merge_opts(RestToAdd, [K | Opts]).